cpp/tests/io/orc_test.cpp

```diff
@@ -1432,7 +1432,7 @@ TEST_F(OrcWriterTest, TestMap)
   auto filepath = temp_env->get_temp_filepath("MapColumn.orc");
   cudf::io::orc_writer_options out_opts =
     cudf::io::orc_writer_options::builder(cudf::io::sink_info{filepath}, expected)
-      .metadata(&expected_metadata);
+      .metadata(expected_metadata);
```

Comment:
Ah, I forgot to reply to @hyperbolic2346 's question about this :(
Since the `metadata` API now takes the object by value, we can avoid a copy if we pass a temporary (rvalue). This can be done even with a variable (lvalue) with the use of `std::move`, which casts the object into rvalue so it can be moved from. 
This move (move ctor, not `std::move`) puts the passed object into an invalid state, so we only use this when the variable is not used afterwards. In some tests we use the metadata object for comparison after round-tripping the table, so I could not use `std::move` there. In tests where the metadata object is not used after it's passed to `metadata()`, I used `std::move` to avoid making a copy.

Reply:
Aha, thanks!